File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/transfer/RequestMoneyTransferCommand.java
package org.jmolecules.example.axonframework.domain.api.command.transfer;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

/**
 * Received by the source account.
 * Reserves amount for transfer and starts saga.
 */
public record RequestMoneyTransferCommand(
  @TargetAggregateIdentifier
  String sourceAccountId,

  String targetAccountId,

  int amount
) {

  // empty

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/core/model/moneytransfer/command/RequestMoneyTransferCommand.kt
package org.jmolecules.example.axonframework.core.model.moneytransfer.command

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.core.model.bankaccount.AccountId
import org.jmolecules.example.axonframework.core.model.bankaccount.Amount
import org.jmolecules.example.axonframework.core.model.moneytransfer.MoneyTransferId

/**
 * Received by the source account.
 * Reserves amount for transfer and starts saga.
 */
@Command(namespace = "axon.bank", name = "RequestMoneyTransferCommand")
data class RequestMoneyTransferCommand(
  @Association
  val sourceAccountId: AccountId,
  val targetAccountId: AccountId,
  val amount: Amount,
  val moneyTransferId: MoneyTransferId
)
